homeassistant/components/smarttub/sensor.py

```diff
@@ -2,17 +2,46 @@
 from enum import Enum
 import logging
 
+import smarttub
+import voluptuous as vol
+
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import config_validation as cv, entity_platform
+
 from .const import DOMAIN, SMARTTUB_CONTROLLER
 from .entity import SmartTubSensorBase
 
 _LOGGER = logging.getLogger(__name__)
 
+
 ATTR_DURATION = "duration"
 ATTR_LAST_UPDATED = "last_updated"
 ATTR_MODE = "mode"
 ATTR_START_HOUR = "start_hour"
 
 
+SET_PRIMARY_FILTRATION_SCHEMA = vol.All(
+    cv.has_at_least_one_key(ATTR_DURATION, ATTR_START_HOUR),
+    cv.make_entity_service_schema(
+        {
+            vol.Optional(ATTR_DURATION): vol.All(int, vol.Range(min=1, max=24)),
+            vol.Optional(ATTR_START_HOUR): vol.All(int, vol.Range(min=0, max=23)),
+        },
+    ),
+)
+
+SET_SECONDARY_FILTRATION_SCHEMA = cv.make_entity_service_schema(
```

Comment:
I think you can omit the `cv.make_entity_service_schema` since it should be wrapped automatically if you pass a bare `dict`